### DISCUSSION TITLE
Using Magic link/passwordless in blitz

### DISCUSSION THREAD

**Discussion by waptik** [blitz-js/blitz#2188] [Ideas]
So in response to [this discord message](https://discord.com/channels/802917734999523368/814628031322128394/828670571776704522)

Here's an integration by @walkaroundcity: https://github.com/walkaroundcity/walkaroundcity/blob/main/integrations/magicLogin.ts
I'm pretty sure it works. 

Sorry for not replying earlier on discord.

**Comment by niklasgrewe** [comment]
@waptik thank you for sharing this kind of authentication. I am also interesting in using magic links within blitzjs. But what happens if I open the Magic Link on the smartphone instead of the PC, will I still be logged into the webapp?

**Reply by waptik** [reply]
You're welcome!
I guess you'll be logged in on the smartphone then. I think the magic link will contain some info that will enable blitz to create user session on the device where the magic link was validated from.

**Reply by niklasgrewe** [reply]
Yes, I think so too, because the request was triggered via the smartphone. Do you have an idea how to get around this, so that you are logged in on the desktop instead of the phone?

**Reply by flybayer** [reply]
@niklasgrewe I don't know of any easy way to do that. It may be possible would be complex and require changes to blitz core

**Reply by niklasgrewe** [reply]
@flybayer ok could you imagine then that such a customization would be possible if SuperTokens also offer Magic Link Login?

**Reply by flybayer** [reply]
We don't use SuperTokens (our custom implementation is just designed by them).

MagicLink works with Blitz, just for the current device you are using.

Does MagicLink work for other devices in other setups?

**Reply by niklasgrewe** [reply]
I have not tried that yet. I'll let you know. Do you know if Passport Magic links are really secure? What would theoretically happen if the link is forwarded. I saw that the tokens are valid for 60min. So could someone foreign log in through the link, or can that be prevented?
